Asterix/wfsc/estimator.py

```diff
@@ -227,47 +185,151 @@ def estimate(self,
+                                                            self.dimEstim,
+                                                            self.amplitudePW,
+                                                            voltage_vector=voltage_vector,
+                                                            wavelengths=wavei)
+
+                    result_estim.append(wfs.calculate_pw_estimate(Difference, self.PWMatrix[i], **kwargs))
+
+            elif self.polychrom == 'centralwl':
+                Difference = wfs.simulate_pw_difference(entrance_EF,
+                                                        testbed,
+                                                        self.posprobes,
+                                                        self.dimEstim,
+                                                        self.amplitudePW,
+                                                        voltage_vector=voltage_vector,
+                                                        wavelengths=testbed.wavelength_0)
+
+                result_estim.append(wfs.calculate_pw_estimate(Difference, self.PWMatrix[0], **kwargs))
+            elif self.polychrom == 'broadband_pwprobes':
+                Difference = wfs.simulate_pw_difference(entrance_EF,
+                                                        testbed,
+                                                        self.posprobes,
+                                                        self.dimEstim,
+                                                        self.amplitudePW,
+                                                        voltage_vector=voltage_vector,
+                                                        wavelengths=testbed.wav_vec)
+                result_estim.append(wfs.calculate_pw_estimate(Difference, self.PWMatrix[0], **kwargs))
+            else:
+                raise Exception(self.polychrom + " is not a valid polychromatic estimation/correction mode")
+            return result_estim
 
         elif self.technique == 'coffee':
             return np.zeros((self.dimEstim, self.dimEstim))
 
         else:
             raise Exception("This estimation algorithm is not yet implemented")
+
+    def find_DM_to_probe(self, testbed: Testbed):
+        """
+            function to find which DM to use for the PW probes
+
+            AUTHOR : Johan Mazoyer
+
+            Parameters
+            ----------
+            testbed :  OpticalSystem.Testbed
+                Testbed object which describe your testbed
+            
+            Returns
+            ----------
+            name_DM_to_probe_in_PW: string
+                name of the DM to probe in PW
+
+
+            """
+
+        # we chose it already. We only check its existence
+        if hasattr(testbed, 'name_DM_to_probe_in_PW'):
+            if testbed.name_DM_to_probe_in_PW not in testbed.name_of_DMs:
+                raise Exception("Cannot use this DM for PW, this testbed has no DM named " +
```

Comment:
This is technically a custom `LookupError` I guess but also here at least a`ValueError` is more specific than just an exception.